### DISCUSSION TITLE
Multiple sitemap.xml file not supported for next js 14

### DISCUSSION THREAD

**Discussion by senthilmca90** [vercel/next.js#58378] [App Router] [answered]
Hi, trying to implement the multiple sitemap.xml file, Next Js is not supported, My sitemap.xml file two large. So, I try to split the file for Google Site Map, sitemap index and each category to change dynamic sitemap file, sitemap1, sitemap2, etc...

Can you please help me out for this one?

**Comment by icyJoseph** [comment, marked as answer]
I think, it'll eventually happen: https://nextjs.org/docs/app/api-reference/file-conventions/metadata/sitemap

<img width="932" alt="Screenshot 2023-11-13 at 13 49 13" src="https://github.com/vercel/next.js/assets/21013447/767eff62-2ab0-42b4-8a43-b408d40a08c5">

In the meantime, you could serve them manually, with a `sitemap.xml/route.ts` route handler, and then you do something like:

```ts
// src/app/<some-path>/sitemap.xml/route.ts
export const dynamic = "force-dynamic";

export async function GET() {
  const xml = "";  // generate your sitemap somehow as xml

  const response = new Response(xml, {
    status: 200,
    statusText: "ok",
  });

  response.headers.append("content-type", "text/xml");

  return response;
}
```

**Comment by programming-with-ia** [comment]
A better sitemap generator for Next.js App Router - full support for images, videos, hreflang alternates, sitemap indexes, custom endpoints, and real caching control with drop-in migration. [View on GitHub](https://github.com/programming-with-ia/better-next-sitemap)
